frontend/src/Judge/reducers.js

```diff
@@ -78,6 +95,34 @@ const queue = (state = initialQueueState, action) => {
 const queues = (state = {}, action) => {
   // Proxy all queue actions to the particular queue we want to target
   switch (action.type) {
+    case actions.FETCH_SUBMISSIONS:
+      const {
+        submissions,
+        username
+      } = action.payload
+      // First, gather the submissions by show ID (they _should_ be all the same
+      // ID anyway, but just in case)
+      const submissionsByShowId = submissions.reduce(
+        (accum, submission) => ({
+          ...accum,
+          [submission.show.id]: [
+            ...(accum[submission.show.id] || []),
+            submission
+          ]
+        }),
+        {}
+      )
+      // proxy the action to each subqueue
+      return Object.entries(submissionsByShowId).reduce(
+        (accum, [showId, submissions]) => ({
+          ...accum,
+          [showId]: queue(
```

Comment:
This section is scoped to one show, right? 

By that I mean, if we view show 1, will only show 1's data be fetched? Then if we view show 2, will only show 2's data be fetched, not fetching both show 1 and show 2's data (i.e. as we look at more shows, not linearly fetching / shuffling more data)?

Reply:
should be.. as above, I can totally modify this so that it assumes you'd only be fetching for one show at once, to clean it up a bit